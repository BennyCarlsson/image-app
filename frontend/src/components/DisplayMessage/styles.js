import styled from 'styled-components'

export const DisplayMessageWrapper = styled.div`
  opacity: ${({ displayMessage }) => (displayMessage ? 1 : 0)};
  display: flex;
  justify-content: center;
  align-items: center;
  height: 48px;
  margin-bottom: 24px;
  width: 100%;
  background-color: #eefaf8;
  font-family: sans-serif;
  font-weight: 700;
  font-size: 1rem;
  color: #373150;
`
