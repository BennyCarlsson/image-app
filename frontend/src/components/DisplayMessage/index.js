import React from 'react'
import { DisplayMessageWrapper } from './styles'

const DisplayMessage = ({ message }) => {
  return (
    <DisplayMessageWrapper displayMessage={!!message} data-testid='message'>
      {message}
    </DisplayMessageWrapper>
  )
}

export default DisplayMessage
