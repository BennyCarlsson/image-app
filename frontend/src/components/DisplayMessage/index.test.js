import React from 'react'
import { render, screen } from '@testing-library/react'
import DisplayMessage from '.'

describe('<DisplayMessage />', () => {
  const message = 'message text'

  it('should match snapshot', () => {
    const { asFragment } = render(<DisplayMessage message={message} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<DisplayMessage message={message} />)
    expect(screen).toBeTruthy()
  })

  it('should render message', () => {
    render(<DisplayMessage message={message} />)
    expect(screen.getByText(message)).toBeInTheDocument()
  })

  it('should render have 0 opacity if no message prop', () => {
    render(<DisplayMessage />)
    expect(screen.getByTestId('message')).toHaveStyle({
      opacity: 0,
    })
  })

  it('should render have 1 opacity if message prop', () => {
    render(<DisplayMessage message={message} />)
    expect(screen.getByText(message)).toHaveStyle({ opacity: 1 })
  })
})
