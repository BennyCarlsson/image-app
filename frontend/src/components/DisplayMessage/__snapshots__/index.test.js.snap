// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayMessage /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gYpFnL"
    data-testid="message"
  >
    message text
  </div>
</DocumentFragment>
`;
